lib/backend/backend.go

```diff
@@ -244,6 +245,27 @@ func NextPaginationKey(r types.Resource) string {
 	return string(nextKey([]byte(r.GetName())))
 }
 
+// MaskKey masks the given key if any of the given sensitive prefixes are detected.
+// If no sensitive prefixes are given, then defaults will be used.
+// e.g "/tokens/123456789" -> "/tokens/******789"
+func MaskKey(key []byte, sensitivePrefixes ...string) []byte {
```

Comment:
I think it's better to keep this unexported in `backend/report.go` and tied with the only current use case. Although you may have some future use cases in mind, unless we need to support them immediately it [might be bad practice](https://en.wikipedia.org/wiki/You_aren't_gonna_need_it) to introduce this generalization at this point.

Also notice how you're splitting and rejoining with `Separator` in `buildKeyLabel` and then splitting and rejoining once again here in `MaskKey` just because the functions are now standalone. Is there anything to outweigh this cumbersome complexity?

Reply:
I found the code to be confusing and was worried about it being misused in the future, so I was trying to separate out the "keylabel" and "keymask" logic and generalize it. I think you're right that this isn't needed right now and instead I might just add some comments to the original code.